dds/idl/itl_generator.cpp

```diff
@@ -285,6 +286,12 @@ bool itl_generator::gen_struct(AST_Structure*, UTL_ScopedName* /* name */,
                   << Open(this)
                   << Indent(this) << "\"kind\" : \"alias\",\n"
                   << Indent(this) << "\"name\" : \"" << repoid << "\",\n"
+
+  // Check if this is defined as a primary data type
+                  << Indent(this) << "\"note\" : { \"is_dcps_data_type\" : "
+                  << ((idl_global->is_dcps_type(name) == 0) ? "false" : "true")
```

Comment:
This could be simplified by swapping true/false so you don't need == 0 and extra parens